HACKING.rst

```diff
@@ -87,6 +87,81 @@ You can also see all the environments by simply running ``tox list``
 Running ``tox run -m format`` and ``tox run -m lint`` before committing code is
 recommended.
 
+Branches
+--------
+
+Starcraft projects follow the `OneFlow`_ git branching model.
+
+
+Branch names
+============
+
+The branch name should be brief and less than 200 characters.
+
+Branch names are formatted as ``<category>/<name>``.
+
+This naming convention provides a few benefits:
+  - GitHub workflows can choose which categories they should run on
+  - GitHub branches rules can be configured per category
+  - Some IDEs and git tools display categorized branches in a neat and
+    nested format
+
+``main``
+########
+
+The main branch containing the latest changes.
+
+``renovate/*``
+##############
+
+Branches created automatically by the `Renovate bot`_.
+
+``work/<ticket>-<description>``
+###############################
+
+For any work driven by a ticketing system, the ticket should be
+part of the branch name. The ticketing system can be built into the repo
+like GitHub issues or an external ticketing system.
+
+For GitHub issue #100 (a ticket to add a README file), the branch would be
+called ``work/100-add-readme``.
+
+For an external ticketing system like Jira, a ticket labeled
+``CRAFT-100`` would use a branch called ``work/CRAFT-100-add-readme``.
+
+``work/<description>``
+######################
+
+For any new features, bug fixes, and general work not driven by a ticketing
+system.
+
+``hotfix/X.Y``
+##############
+
+For hotfixes to an existing minor release.
```

Comment:
Some sections are purposefully light on details to avoid bike-shedding on the OneFlow strategy.  I don't want to get into those specifics in this PR.